e2e/support/helpers/e2e-custom-column-helpers.ts

```diff
@@ -52,3 +63,162 @@ export function checkExpressionEditorHelperPopoverPosition() {
     });
   });
 }
+
+export const CustomExpressionEditor = {
+  get() {
+    return cy.findByTestId("custom-expression-query-editor");
+  },
+  /**
+   * @param {string} text the formula to type
+   * @param {Object} options
+   * @param {boolean} [options.focus]
+   *   true by default. Set to false to stop the helper from explicitly focussing the formula input.
+   * @param {boolean} [options.allowFastSet]
+   *   true by default. However, if you need to examine the popover in the test, it should be set to false so the popover is not dismissed
+   *   Because CodeMirror uses a contenteditable div, and it is not possible to use cy.type() on it, we emulate .type with realPress.
+   *   This does not always work, since realPress() does not support all characters. Setting this to true will enable an escape hatch
+   *   that uses cy.invoke('text') under the hood, to allow for formulas that contain unsupported characters.
+   *   This has some other sideeffects however, so use it sparingly.
+   */
+  type(
+    text: string,
+    {
+      allowFastSet = false,
+      focus = true,
+    }: {
+      focus?: boolean;
+      allowFastSet?: boolean;
+    } = {},
+  ) {
+    if (focus) {
+      CustomExpressionEditor.focus();
+    }
+
+    if (allowFastSet) {
+      // Enter the formula in one go
+      // HACK: we do invoke("text") instead of type() because type() does not work on
+      // CodeMirror elements in Cypress. realType() would work but some of the formulas
+      // contain special characters that are not supported by realType().
+      CustomExpressionEditor.get().findByRole("textbox").invoke("text", text);
+      return CustomExpressionEditor;
+    }
+
+    const parts = text.replaceAll("{{", "{{}{{}").split(/(\{[^}]+\})/);
+
+    parts.forEach(part => {
+      switch (part.toLowerCase()) {
```

Comment:
This is duplicated from the Native Expression editor, I will clean it up in a follow up PR.